addons/pos_viva_wallet/static/src/app/services/pos_store.js

```diff
@@ -12,4 +12,54 @@ patch(PosStore.prototype, {
             }
         });
     },
+
+    async pay() {
+        const currentOrder = this.getOrder();
+        const refundedOrder = currentOrder?.lines[0]?.refunded_orderline_id?.order_id;
+        const vivaWalletPaymentMethod = currentOrder.config_id.payment_method_ids.find(
+            (pm) => pm.use_payment_terminal === "viva_wallet"
+        );
+        await super.pay();
```

Comment:
```suggestion
        await super.pay();
        const currentOrder = this.getOrder();
        const refundedOrder = currentOrder?.lines[0]?.refunded_orderline_id?.order_id;
        const vivaWalletPaymentMethod = currentOrder.config_id.payment_method_ids.find(
            (pm) => pm.use_payment_terminal === "viva_wallet"
        );
```

don't know about return of `super()` check it!